includes/wikia/services/WikiService.class.php

```diff
@@ -47,66 +52,73 @@ class WikiService extends WikiaModel {
-		wfProfileIn( __METHOD__ );
+		if ( !empty( $this->wg->FounderEmailsDebugUserId ) ) {
+			return [ $this->wg->FounderEmailsDebugUserId ];
+		}
 
-		$userIds = array();
-		if ( empty($this->wg->FounderEmailsDebugUserId) ) {
-			// get founder
-			$wikiId = ( empty($wikiId) ) ? $this->wg->CityId : $wikiId ;
-			$wiki = WikiFactory::getWikiById($wikiId);
-			if ( !empty($wiki) && $wiki->city_public == 1 ) {
-				if ($includeFounder) {
-					$userIds[] = $wiki->city_founding_user;
-				}
+		$wikiId = empty( $wikiId ) ? $this->wg->CityId : $wikiId ;
+		$wiki = WikiFactory::getWikiById( $wikiId );
 
-				// get admin and bureaucrat
-				if ( empty($this->wg->EnableAnswers) ) {
-					$memKey = $this->getMemKeyAdminIds( $wikiId, $excludeBots, $limit );
-
-					$adminIds = WikiaDataAccess::cache(
-						$memKey,
-						60 * 60 * 3,
-						function() use ($wiki, $useMaster, $excludeBots, $limit) {
-							$dbname = $wiki->city_dbname;
-							$dbType = ( $useMaster ) ? DB_MASTER : DB_SLAVE;
-							$db = wfGetDB( $dbType, array(), $dbname );
-
-							$conditions = array("ug_group in ('sysop','bureaucrat')");
-
-							if ($excludeBots) {
-								$conditions[] = "ug_user not in (select distinct ug_user from user_groups where ug_group in (" .
-									$db->makeList(self::$botGroups) .
-									"))";
-							}
-
-							$result = $db->select(
-								'user_groups',
-								'distinct ug_user',
-								$conditions,
-								__METHOD__,
-								(!empty($limit))?(array('LIMIT' => $limit)):array()
-							);
-
-							$adminIds = array();
-							while ( $row = $db->fetchObject($result) ) {
-								$adminIds[] = $row->ug_user;
-							}
-							$db->freeResult( $result );
-							return $adminIds;
-						}
-					);
+		if ( empty( $wiki ) || $wiki->city_public != 1 ) {
+			return [];
+		}
 
-					$userIds = array_unique( array_merge($userIds, $adminIds) );
-				}
-			}
-		} else {
-			$userIds[] = $this->wg->FounderEmailsDebugUserId;
+		// Get founder
+		$userIds = [];
+		if ( $includeFounder ) {
+			$userIds[] = $wiki->city_founding_user;
 		}
 
-		wfProfileOut( __METHOD__ );
+		if ( !empty( $this->wg->EnableAnswers ) ) {
+			return $userIds;
+		}
+
+		// Get admins and bureaucrats
+		$memKey = $this->getMemKeyAdminIds( $wikiId, $excludeBots, $limit );
+		$adminIds = WikiaDataAccess::cache(
+			$memKey,
+			self::WIKI_ADMIN_IDS_CACHE_TTL,
+			function() use ( $wiki, $useMaster, $excludeBots, $limit ) {
+				$dbName = $wiki->city_dbname;
+				$dbType = $useMaster ? DB_MASTER : DB_SLAVE;
+				$db = wfGetDB( $dbType, [], $dbName );
+
+				return self::getAdminIdsFromDB( $db, $excludeBots, $limit );
+			}
+		);
+
+		$userIds = array_unique( array_merge( $userIds, $adminIds ) );
+
 		return $userIds;
 	}
 
+	private static function getAdminIdsFromDB( DatabaseBase $db, $excludeBots = false, $limit = null ) {
+		$conditions = [ "ug_group in ('sysop','bureaucrat')" ];
```

Comment:
Yah, this was a copy paste rearrangement of the old code but we should update this.